python/cudf/cudf/core/dataframe.py

```diff
@@ -6419,9 +6419,14 @@ def append(
                 "verify_integrity parameter is not supported yet."
             )
 
-        if isinstance(other, (cudf.Series, dict)):
-            if isinstance(other, dict):
-                other = Series(other)
+        if isinstance(other, dict):
+            if not ignore_index:
+                raise TypeError("Can only append a dict if ignore_index=True")
+            other = DataFrame(other)
+            result = cudf.concat([self, other], ignore_index=False, sort=sort)
```

Comment:
Can we just set `ignore_index=True` here since we need to throw away the existing index of `self` and return a new RangeIndex anyway?

Reply:
This was necessary because we currently end up ignoring the index completely if `ignore_index` is True here: https://github.com/rapidsai/cudf/blob/branch-0.15/python/cudf/cudf/_lib/concat.pyx#L46

This will not work in cases where we have empty data frames but have index set.

Reply:
That feels like a bug in concatenate then, where we should handle dataframes with no columns but have an index of size N and concatenate with a dataframe with some columns, we need to appropriately fill with nulls as expected.

Reply:
i.e.
```
In [153]: pd.concat([pd.DataFrame(index=['a', 'b', 'c']), pd.DataFrame({'a': [1, 2, 3]})], ignore_index=True)
Out[153]:
     a
0  NaN
1  NaN
2  NaN
3  1.0
4  2.0
5  3.0
```

Reply:
Seems like we'll need to make few changes in the `concat` code-flow, so thought I'll handle in a separate PR first and get it in before this PR. 

issue: https://github.com/rapidsai/cudf/issues/5639

Reply:
This is now fixed and synced up.